### PR TITLE
ButtonGroup Component

### PR DESCRIPTION
**Pull request by antictor**
## Related issue
#219

The button group accepts a list of button item properties and renders a group of like styled buttons as a unit.

Oct-30-2019 10-10-31

### CODE REVIEW

File: packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import React from 'react'
+
+export const ButtonGroup: React.FC = ({ children }) => (
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx]: Is there an opportunity to tighten this up with an interface as `children` will always be a collection of `Button`?

File: packages/docs-site/src/library/pages/components/buttongroup.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
+---
+title: Button Group
+description: A collection of buttons commonly used to select an option
+header: true
+---
+
+import { Button,ButtonGroup,  Icons, Tab, TabSet } from '@royalnavy/react-component-library'
```
Comment (thyhjwb6) [inline packages/docs-site/src/library/pages/components/buttongroup.md]: Spacing could be tweaked here

File: packages/docs-site/src/library/pages/components/buttongroup.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+### Sizing & Spacing
+
+
+### Hierarchy & Placement
+
+</Tab>
+
+<Tab title="Develop">
+
+A ButtonGroup is a simple wrapper component that is designed to group buttons together into a common component. The ButtonGroup component itself accept no properties, only content which should consist of a collection of buttons.
```
Comment (thyhjwb6) [inline packages/docs-site/src/library/pages/components/buttongroup.md]: `accept` 👉 `accepts`

**Review by thyhjwb6 — COMMENTED** [review]
A few small comments

**Comment by hxltrhuxze** [comment]
Several feedback points:
1. The button border needs to be from the button itself. This PR is adding a neutral border to the first child for some reason.
2. I'd prefer it if all the buttons inside the group were forced to be the same type. Having a mix of primary/danger/neutral is confusing.
3. what's going on with this button? We have dropdowns which != buttons, so I'm not sure why this button has a downwards icon here. Also, why is the icon blue?
Screenshot 2019-10-29 at 15 23 20

File: packages/docs-site/src/library/pages/components/buttongroup.md (hunk 2 above)
Comment (antictor) [reply packages/docs-site/src/library/pages/components/buttongroup.md]: This file is still being worked on by Ed.

Comment (antictor) [reply packages/docs-site/src/library/pages/components/buttongroup.md]: Waiting for design documentation to be added

File: packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx (hunk 1 above)
Comment (antictor) [reply packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx]: Could do. Whats the syntax for that?

Comment (antictor) [reply packages/react-component-library/src/components/ButtonGroup/ButtonGroup.tsx]: NVM if other comments are correct from design then I may have to write this a different way

**Comment by antictor** [comment]
The dropdown icon was just to show you can have an icon and I had that icon already in the project. Could be any icon

To force the buttons to all be the same type may be restrictive. There may be a time when one of the buttons has to be red surely?

If not then that and the border thing will mean a re-think of the approach. Instead of a simple wrapper I'd have to pass a single size, type, variant and an array of labels/keys, and expect an onClick handler that would be triggered with the key of the button pressed.

Not saying it's wrong, just clarifying before I continue

**Comment by hxltrhuxze** [comment]
> The dropdown icon was just to show you can have an icon and I had that icon already in the project. Could be any icon.
Fair point, let's not use that example in the docs though. Why is there space to the right of the close button, just out of curiosity? It should be equally spaced.

> To force the buttons to all be the same type may be restrictive. There may be a time when one of the buttons has to be red surely?

Maybe it is too restrictive. However, I do feel that from a UI perspective that having a split between multiple types of actions prevents a potential mess resulting from this component. 

Primary buttons are your main action and should be standalone - they take prominence in the interface and are _the_ thing we want the user to focus on. Button groups are there to supply complimentary actions under one theme. We're always aiming to keep the number of primary buttons per component/page low, as when we start using more of them, they lose their importance. 

I can't help but think having [Neutral][Neutral][Danger][Neutral] is a really bad UX practice, as you have a destructive action buried in between regular buttons. There's no hierarchy of responsibility there.

Regardless, I think we should force button groups to be purely Secondary buttons. The buttons in the groups are all of equal prominence.

> If not then that and the border thing will mean a re-think of the approach. Instead of a simple wrapper I'd have to pass a single size, type, variant and an array of labels/keys, and expect an onClick handler that would be triggered with the key of the button pressed.

You'd need to pass a size anyway as you can't mix multiple sized buttons inside a button group. The original design issue (#219) has the button example there. It's all using secondary buttons - there's no primary buttons in here. 

Either way, what we do to the hierarchy is irrelevant to the border issue. There's a neutral border being set on the internal buttons - why? If you wanted to make the button group accept any type of button, why is there a neutral border being set? It literally _isn't_ what the border is on any of the primary buttons, so it's going to look visually off regardless of which buttons are supplied.

**Comment by antictor** [comment]
Oct-31-2019 10-43-06

**Review by m7kvqbe1 — APPROVED** [review]

**Pull request merged**